azuredevops/internal/service/git/resource_git_repository.go

```diff
@@ -167,6 +191,10 @@ func resourceGitRepositoryCreate(d *schema.ResourceData, m interface{}) error {
 		return fmt.Errorf("Error creating repository in Azure DevOps: %+v", err)
 	}
 
+	// set the id immediately after successfully creating the repository, which will allow terraform to track the
+	// repository in state and will taint the resource if further errors are encountered during initialization
+	d.SetId(createdRepo.Id.String())
```

Comment:
Not sure how maintainers will feel about this, so I wanted to point it out. The benefit here is that terraform will taint the resource if an error is returned, which ensures that manual intervention is never required in the event of a failure. Any subsequent apply will recreate the resource. Prior to implementing this, I tried using a source url that had embedded basic auth username:password@, which failed to import and the repository was left behind requiring manual cleanup in order to run terraform again -- this resolves that issue.